Source/WebCore/page/text-extraction/TextExtraction.cpp

```diff
@@ -1773,5 +1784,105 @@ std::optional<SimpleRange> rangeForExtractedText(const LocalFrame& frame, Extrac
     return searchForText(*node, text);
 }
 
+Vector<FilterRule> extractRules(Vector<FilterRuleData>&& data)
+{
+    return WTFMove(data).map([](auto&& data) -> FilterRule {
+        auto&& [name, urlPattern, scriptSource] = WTFMove(data);
+        if (urlPattern.isEmpty())
+            return { WTFMove(name), std::nullopt, WTFMove(scriptSource) };
+
+        auto regex = Yarr::RegularExpression { urlPattern, { Yarr::Flags::IgnoreCase } };
+        if (!regex.isValid())
+            return { WTFMove(name), std::nullopt, WTFMove(scriptSource) };
+
+        return { WTFMove(name), { WTFMove(regex) }, WTFMove(scriptSource) };
+    });
+}
+
+static DOMWrapperWorld& filteringWorld()
+{
+    static NeverDestroyed<RefPtr<DOMWrapperWorld>> world;
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [&] {
+        world.get() = DOMWrapperWorld::create(commonVM(), DOMWrapperWorld::Type::Internal, "Text Extraction Filtering Rules"_s);
+    });
```

Comment:
We have thread-safe statics enabled in WebCore now, so I think you can drop the `std::call_once`? https://github.com/WebKit/WebKit/commit/2d4ca326601eb0a377673f8200349913aae83c65




```suggestion
    static NeverDestroyed<RefPtr<DOMWrapperWorld>> world = DOMWrapperWorld::create(commonVM(), DOMWrapperWorld::Type::Internal, "Text Extraction Filtering Rules"_s);
```

Reply:
Good point — removed the `call_once` here. Thanks for the review!